Library/Homebrew/cmd/update.sh

```diff
@@ -448,7 +426,38 @@ EOS
   for DIR in "$HOMEBREW_REPOSITORY" "$HOMEBREW_LIBRARY"/Taps/*/*
   do
     [[ -d "$DIR/.git" ]] || continue
-    pull "$DIR"
+
+    # Used for testing purposes, e.g., for testing formula migration after
+    # renaming it in the currently checked-out branch. To test run
+    # "brew update --simulate-from-current-branch"
+    if [[ -n "$HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH" ]]
+    then
+      UPSTREAM_BRANCH="$(upstream_branch)"
+      INITIAL_REVISION="$(git rev-parse -q --verify "$UPSTREAM_BRANCH")"
+      CURRENT_REVISION="$(read_current_revision)"
+      export HOMEBREW_UPDATE_BEFORE"$TAP_VAR"="$INITIAL_REVISION"
+      export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$CURRENT_REVISION"
+      if [[ "$INITIAL_REVISION" != "$CURRENT_REVISION" ]]
+      then
+        HOMEBREW_UPDATED="1"
+      fi
+      if ! git merge-base --is-ancestor "$INITIAL_REVISION" "$CURRENT_REVISION"
+      then
+        odie "Your $DIR HEAD is not a descendant of $UPSTREAM_BRANCH!"
+      fi
+      continue
+    fi
+
+    TAP_VAR=$(repo_var "$DIR")
+    if [[ -n "$(find "$DIR/.git/FETCH_HEAD" -type f -mtime -60s)" ]]
```

Comment:
Is this a good indicator that something was fetched? Would comparing the `origin/master` commit hash before and after the fetch maybe be a better indicator? (Not sure what the best solution is, but this feels like a strange criterion. It also impacts the non-`--preinstall` case which it probably shouldn't?)

However, if we stick with this, this should get a `2>/dev/null` just in case.